### ISSUE TITLE
DeprecationWarning: current URL string parser is deprecated

### ISSUE THREAD

**Issue by amenadiel** [strongloop/loopback-connector-mongodb#442]
# Description/Steps to reproduce

Running a loopback instance with `"loopback-connector-mongodb": "^3.4.4"` I get a deprecation warning:

```sh
DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect
```

Which in turn comes from a new valid option detailed in:

https://github.com/mongodb/node-mongodb-native/blob/v3.1.0/lib/mongo_client.js#L112

```js
* @param {boolean} [options.useNewUrlParser=false] Determines whether or not to use the new url parser
```
Also, this key has been added to `validOptionNames`  at

https://github.com/mongodb/node-mongodb-native/blob/v3.1.0/lib/operations/mongo_client_ops.js#L102

**However** if I manually add that option at 

https://github.com/strongloop/loopback-connector-mongodb/blob/205a3a35c3dde548f9b98af199b5edfab4d459f0/lib/mongodb.js#L145-L148

I get a ton of errors, basically because this "new url parser" doesn't infer the `replicaSet` parameter from the connection string.

So this is just a heads up to see if in a near future we'll need to introduce changes to this connector.

**Comment by ingkenai** [comment]
to avoid that warning I stay in this version: "mongoose": "5.0.10",  for a while

**Comment by 07ASHISH** [comment]
to avoid this warning you have to pass in connect function { useNewUrlParser: true } this will remove that 
 error 
MongoClient.connect(url, { useNewUrlParser: true } );

**Comment by azicchetti** [comment]
I can't connect to mlab with the 3.4.4 connector.
How can I fix that other than downgrading the package?

**Comment by npn225** [comment]
ingkenai's solution works!  

For MEAN Stack users: 

1) Go to "package.json" file and change the mongoose version to "5.0.10"

2) Save this change to the "package.json" file

3) Head to the command prompt and make sure you are NOT running the app using "node" or "nodemon" commands!  

4) Enter and run the command "npm i" in order to install the packages associated to the changes you made in the "package.json" file

5) Then use "node" or "nodemon" commands to run the API without depreciation warning error!

**Comment by niranjanreddy891** [comment]
Have anyone found solution for this issue?

**Comment by npn225** [comment]
Don't know why you thumbed down my comment niranjanreddy891!

**Comment by niranjanreddy891** [comment]
No. The solution which you gave doesn't help me.

**Comment by npn225** [comment]
What exactly did you do when implementing my steps to the solution?  

Are you even using the MEAN stack?

Can you show pictures of the things you did?

**Comment by niranjanreddy891** [comment]
I got my own solution @npn225.

**Comment by npn225** [comment]
@niranjanreddy891 

That's great!  

You can post your solution here so that anyone in your situation will know what to do and how to fix the problem!  

The least any of us can do is contribute to the community!

**Comment by niranjanreddy891** [comment]
`The least any of us can do is contribute to the`

**Comment by niranjanreddy891** [comment]
`make -k build-with-full-infra`

**Comment by matteo-pennisi** [comment]
To fix the problem upgrade loopback-connector-mongodb (I'm using 3.6.0)
and add this to your MongoDB data source in datasources.json : ` "useNewUrlParser":"true"`
For example now my connector looks like this:
```
"mongodb": {
      "url": "myDBurl",
      "name": "mongodb",
      "connector": "loopback-connector-mongodb",
      "useNewUrlParser":"true"
    }
```

**Comment by niranjanreddy891** [comment]
I solved in the same way as @matteo-pennisi mentioned. But I was busy at work. So, I replied late. Sorry for that :)

**Comment by virkt25** [comment]
Closing this issue as fixed. The change was made in https://github.com/strongloop/loopback-connector-mongodb/pull/450 and released in version 3.6.0 as mentioned by @niranjanreddy891

**Comment by idindrakusuma** [comment]
> To fix the problem upgrade loopback-connector-mongodb (I'm using 3.6.0)
> and add this to your MongoDB data source in datasources.json : `"useNewUrlParser":"true"`
> For example now my connector looks like this:
> 
> ```
> "mongodb": {
>       "url": "myDBurl",
>       "name": "mongodb",
>       "connector": "loopback-connector-mongodb",
>       "useNewUrlParser":"true"
>     }
> ```

Thank you :)
